Update step-40.

drwells opened the pull request:
Part of #6774.

drwells commented on examples/step-40/step-40.cc:
```diff
@@ -693,17 +690,27 @@ namespace Step40
+// order (i.e., the reverse of the order of construction). The last function
+// called by Utilities::MPI::MPI_InitFinalize::~MPI_InitFinalize() is
+// <code>MPI_Finalize</code>: i.e., once this object is destructed the program
+// should exit since MPI will no longer be available.
```
I couldn't figure out what the original text meant by 'pair of braces' so I just rewrote the whole thing.

masterleinad replied:
Sounds good!

masterleinad commented on examples/step-40/step-40.cc:
```diff
@@ -403,19 +400,18 @@ namespace Step40
-                    (rhs_value * fe_values.shape_value(i, q_point) *
-                     fe_values.JxW(q_point));
+                  cell_rhs(i) += rhs_value * fe_values.shape_value(i, q_point) *
+                                 fe_values.JxW(q_point);
```
Should we split this into three lines?

drwells replied:
We do that everywhere else so yes, lets do it here too.

drwells commented on examples/step-40/step-40.cc:
```diff
@@ -280,13 +280,13 @@ namespace Step40
-    // are the locally relevant ones, so we pass this to the
-    // ConstraintMatrix::reinit function. As a side note, if you forget to
-    // pass this argument, the ConstraintMatrix class will allocate an array
+    // consequence, we need to tell the affine constraints object for which
```
I agree.

drwells commented on examples/step-40/step-40.cc:
```diff
@@ -280,13 +280,13 @@ namespace Step40
+    // consequence, we need to tell the affine constraints object for which
+    // degrees of freedom it can store constraints and for which it may not
+    // expect any information to store. In our case, as explained in the @ref
+    // distributed module, the degrees of freedom we need to care about on
```
It took me a few minutes to figure out what you meant. I fixed this (and checked the other ones).

bangerth replied:
I think you need to put the `@ref` onto the same line as the tag.

drwells replied:
@bangerth I am not sure that I understand your comment. I think I fixed all of them in a subsequent commit: am I missing one?

masterleinad replied:
I believe he was looking at an outdated version.

drwells replied:
That seems likely :)

bangerth replied:
Ah, correct, yes. Always behind the times these days -- reviewing and merging dozens of ASPECT pull requests a day distracts a lot from looking at deal.II stuff :-)

drwells commented:
I fixed everything that was commented on. I also added some more text that clarifies that one can use either PETSc or Trilinos (and, subsequently, either Hypre or ML).

masterleinad commented:
/run-tests
